### PR TITLE
Update pypy

### PR DESCRIPTION
**Pull request by lmctv**
Instead of staying on pypy (and cffi) versions from october/november 2015

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -46,8 +46,8 @@
             "upgrade PyPy to use this library."
         )
 else:
-    requirements.append("cffi>=1.1.0")
-    setup_requirements.append("cffi>=1.1.0")
+    requirements.append("cffi>=1.7.0")
```
Comment (reaperhulk) [inline setup.py]: Bumping the cffi requirement to require the latest without a reason (e.g. we need a feature only provided by a newer version) is not something we want to do.

**Comment by reaperhulk** [comment]
If you remove that cffi bump we can merge the pypy upgrade though!

File: setup.py [hunk 2]
```diff
@@ -46,8 +46,8 @@
             "upgrade PyPy to use this library."
         )
 else:
-    requirements.append("cffi>=1.7.0")
-    setup_requirements.append("cffi>=1.7.0")
+    requirements.append("cffi>=1.4.1")
+    setup_requirements.append("cffi>=1.4.1")
```
Comment (lmctv) [inline setup.py]: I think this could help in correctly solving #186 insetead of working around the problem like in #187

File: setup.py [hunk 3]
```diff
@@ -46,8 +46,8 @@
             "upgrade PyPy to use this library."
         )
 else:
-    requirements.append("cffi>=1.1.0")
-    setup_requirements.append("cffi>=1.1.0")
+    requirements.append("cffi>=1.4.1")
```
Comment (reaperhulk) [inline setup.py]: Why the increase to 1.4.1?

**Comment by lmctv** [comment]
OK: I removed the cffi minimum version requirement and will put the new requirement when I come to the init_once PR.

**Pull request merged**